Library/include/CSP/Systems/SystemBase.h

```diff
@@ -25,17 +26,52 @@ class WebClient;
 
 }
 
+namespace csp::multiplayer
+{
+
+class EventBus;
+class EventDeserialiser;
+class MultiplayerConnection;
+
+} // namespace csp::multiplayer
 
 namespace csp::systems
 {
-/// @brief Base class for all Connected Spaces Platform Systems, which enforces passing of a WebClient instance in the constructor of each System.
+/// @brief Base class for all Connected Spaces Platform Systems, which enforces passing of a WebClient or EventBus instance in the constructor of each
+/// System.
 class CSP_API CSP_NO_DISPOSE SystemBase
 {
+	friend class csp::multiplayer::MultiplayerConnection;
+
 protected:
-	SystemBase() : WebClient(nullptr) {};
-	CSP_NO_EXPORT SystemBase(csp::web::WebClient* InWebClient) : WebClient(InWebClient) {};
+	CSP_NO_EXPORT SystemBase(csp::web::WebClient* InWebClient, csp::multiplayer::EventBus* InEventBus);
+	CSP_NO_EXPORT SystemBase(csp::multiplayer::EventBus* InEventBus);
 
 	csp::web::WebClient* WebClient;
+	csp::multiplayer::EventBus* EventBusPtr;
+
+public:
+	/// @brief Destructor of the SystemBase base class.
+	virtual ~SystemBase();
+
+	/// @brief Registers the system to listen for the default event.
+	virtual void RegisterSystemCallback();
+
+	/// @brief Deregisters the system from listening for the default event.
+	virtual void DeregisterSystemCallback();
+
+	/// @brief Deserialises the event values of the system.
+	/// @param EventValues std::vector<signalr::value> : event values to deserialise
+	CSP_NO_EXPORT virtual void Deserialise(const std::vector<signalr::value>& EventValues);
```

Comment:
Would this make more sense to be called "OnEvent", as we dont know what the system is going to do wth the event

Reply:
Could do, I don't have a preference, @MAG-AdamThorn do you?

Reply:
@MAG-AdamThorn In the context of the event bus sending the event to the system, imo "OnEvent" feels more clear

Reply:
Sorry missed this. Yes I think OnEvent makes more sense here.